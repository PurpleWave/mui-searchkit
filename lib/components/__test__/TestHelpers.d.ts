export declare const fastClick: (el: any) => void;
