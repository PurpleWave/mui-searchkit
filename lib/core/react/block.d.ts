/**
 * Entry point
 * @param {String} name Block name
 * @return {Function}
 */
export declare function block(name?: any): any;
