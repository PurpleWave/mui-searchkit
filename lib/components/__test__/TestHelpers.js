Object.defineProperty(exports, "__esModule", { value: true });
exports.fastClick = function (el) {
    el.simulate("click");
};
//# sourceMappingURL=TestHelpers.js.map