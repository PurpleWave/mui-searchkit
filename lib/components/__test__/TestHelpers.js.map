{"version":3,"file":"TestHelpers.js","sourceRoot":"","sources":["../../../src/components/__test__/TestHelpers.ts"],"names":[],"mappings":";AAAa,QAAA,SAAS,GAAG,UAAC,EAAE;IAC1B,EAAE,CAAC,QAAQ,CAAC,OAAO,CAAC,CAAA;AACtB,CAAC,CAAA"}