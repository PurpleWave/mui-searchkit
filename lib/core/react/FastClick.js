var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var RenderComponent_1 = require("./RenderComponent");
var NormalClickComponent = /** @class */ (function (_super) {
    __extends(NormalClickComponent, _super);
    function NormalClickComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NormalClickComponent.prototype.render = function () {
        var children = this.props.children;
        return React.cloneElement(children, {
            onClick: this.props.handler
        });
    };
    return NormalClickComponent;
}(React.PureComponent));
exports.NormalClickComponent = NormalClickComponent;
var FastClickComponent = /** @class */ (function (_super) {
    __extends(FastClickComponent, _super);
    function FastClickComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.threshold = 20;
        return _this;
    }
    FastClickComponent.prototype.handleMouseDown = function (event) {
        if (this.supportsTouch)
            return;
        if (event.button === 0) {
            this.props.handler();
        }
    };
    FastClickComponent.prototype.cleanupTouch = function () {
        delete this.startPoint;
    };
    FastClickComponent.prototype.getSinglePoint = function (event) {
        var touches = event.changedTouches;
        if (touches.length === 1) {
            return {
                x: touches[0].pageX,
                y: touches[0].pageY
            };
        }
        return null;
    };
    FastClickComponent.prototype.handleTouchStart = function (event) {
        this.supportsTouch = true;
        this.startPoint = this.getSinglePoint(event);
    };
    FastClickComponent.prototype.pointsWithinThreshold = function (p1, p2) {
        return (Math.abs(p1.x - p2.x) < this.threshold &&
            Math.abs(p1.y - p2.y) < this.threshold);
    };
    FastClickComponent.prototype.handleTouchEnd = function (event) {
        if (this.startPoint) {
            var endPoint = this.getSinglePoint(event);
            if (this.pointsWithinThreshold(this.startPoint, endPoint)) {
                this.props.handler();
            }
            this.cleanupTouch();
        }
    };
    FastClickComponent.prototype.handleClick = function (event) {
        event.preventDefault();
    };
    FastClickComponent.prototype.render = function () {
        var children = this.props.children;
        return React.cloneElement(children, {
            onMouseDown: this.handleMouseDown.bind(this),
            onTouchStart: this.handleTouchStart.bind(this),
            onTouchEnd: this.handleTouchEnd.bind(this),
            onClick: this.handleClick.bind(this)
        });
    };
    return FastClickComponent;
}(React.PureComponent));
exports.FastClickComponent = FastClickComponent;
var FastClick = /** @class */ (function (_super) {
    __extends(FastClick, _super);
    function FastClick() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FastClick.prototype.render = function () {
        return RenderComponent_1.renderComponent(FastClick.component, this.props, this.props.children);
    };
    FastClick.component = NormalClickComponent;
    return FastClick;
}(React.Component));
exports.FastClick = FastClick;
//# sourceMappingURL=FastClick.js.map