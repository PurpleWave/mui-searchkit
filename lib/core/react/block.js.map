{"version":3,"file":"block.js","sourceRoot":"","sources":["../../../src/core/react/block.ts"],"names":[],"mappings":";AAAA,IAAM,MAAM,GAAG,OAAO,CAAC,eAAe,CAAC,CAAA;AACtC,4BAA4B;AAE5B,IAAI,EAAE,GAAG,KAAK,EACb,KAAK,GAAG,GAAG,EACX,QAAQ,GAAG;IACV,EAAE,EAAE,EAAE;IACN,EAAE,EAAE,IAAI;IACR,GAAG,EAAE,GAAG;IACR,QAAQ,EAAE,GAAG;IACb,QAAQ,EAAE,EAAE;CACZ,CAAC;AAEH;;GAEG;AACH,gBAAgB,MAAM,EAAE,GAAG;IAC1B,OAAO,MAAM,CAAC,IAAI,CAAC,GAAG,CAAC,CAAC,MAAM,CAAC,UAAS,MAAM,EAAE,GAAG;QAClD,IAAI,KAAK,GAAG,GAAG,CAAC,GAAG,CAAC,CAAC;QAErB,wBAAwB;QACxB,IAAK,KAAK,CAAC,OAAO,CAAC,KAAK,CAAC,EAAG;YAC3B,KAAK,GAAG,KAAK,CAAC,KAAK,EAAE,CAAC;SACtB;QAED,MAAM,CAAC,GAAG,CAAC,GAAG,KAAK,CAAC;QACpB,OAAO,MAAM,CAAC;IACf,CAAC,EAAE,MAAM,CAAC,CAAC;AACZ,CAAC;AAED;;GAEG;AACH,cAAc,GAAG;IAChB,OAAO,MAAM,CAAC,EAAE,EAAE,GAAG,CAAC,CAAC;AACxB,CAAC;AAED;;;;;;GAMG;AACH,uBAAuB,GAAG,EAAE,SAAU;IACrC,IAAK,SAAS,KAAK,SAAS,EAAG;QAC9B,SAAS,GAAG,QAAQ,CAAC,GAAG,CAAC;KACzB;IAED,IAAI,iBAAiB,GAAG,QAAQ,CAAC,QAAQ,CAAC;IAE1C,OAAO,MAAM,CAAC,IAAI,CAAC,GAAG,CAAC,CAAC,MAAM,CAAC,UAAS,KAAK,EAAE,GAAG;QACjD,IAAI,KAAK,GAAG,GAAG,CAAC,GAAG,CAAC,CAAC;QAErB,IAAK,CAAC,KAAK,EAAG;YACb,OAAO,KAAK,CAAC;SACb;QAED,IAAK,KAAK,KAAK,IAAI,EAAG;YACrB,KAAK,CAAC,IAAI,CAAC,SAAS,GAAG,GAAG,CAAC,CAAC;SAC5B;aAAM;YACN,kDAAkD;YAClD,KAAK,CAAC,IAAI,CAAC,SAAS,GAAG,GAAG,GAAG,iBAAiB,GAAG,KAAK,CAAC,CAAC;SACxD;QAED,OAAO,KAAK,CAAC;IACd,CAAC,EAAE,EAAE,CAAC,CAAC;AACR,CAAC;AAED;;;;GAIM;AACN,0BAA0B,IAAI;IAC7B,OAAO,QAAQ,CAAC,QAAQ,CAAC,IAAI,CAAC,IAAI,IAAI,CAAC;AACxC,CAAC;AAED;;GAEG;AACH;IACC,IAAI,IAAI,GAAG,KAAK,CAAC,SAAS,CAAC,KAAK,CAAC,IAAI,CAAC,SAAS,CAAC,EAC/C,OAAO,GAAG,IAAI,CAAC,IAAI,CAAC,CAAC;IAEtB,OAAO,GAAG,IAAI,CAAC,MAAM,CAAC,UAAS,OAAO,EAAE,IAAI;QAC3C,IAAK,IAAI,IAAI,OAAO,IAAI,KAAK,QAAQ,EAAG;YACvC,OAAO,CAAC,IAAI,GAAG,OAAO,CAAC,IAAI,GAAG,QAAQ,CAAC,EAAE,GAAG,IAAI,CAAC;SACjD;QAED,IAAK,IAAI,IAAI,OAAO,IAAI,KAAK,QAAQ,EAAG;YACvC,OAAO,CAAC,IAAI,CAAC,IAAI,CAAC,IAAI,CAAC,CAAC;SACxB;QAED,OAAO,OAAO,CAAC;IAChB,CAAC,EAAE,OAAO,CAAC,CAAC;IAEZ,OAAO,OAAO,CAAC,OAAO,CAAC,CAAC;AACzB,CAAC;AAED;;GAEG;AACH;IACC,gBAAgB;IAChB,IAAI,IAAI,GAAG,QAAQ,CAAC,EAAE,GAAG,IAAI,CAAC,IAAI,EACjC,SAAS,GAAG,gBAAgB,CAAC,IAAI,CAAC,CAAC;IAEpC,gBAAgB;IAChB,SAAS,GAAG,IAAI,CAAC,IAAI,CAAC,MAAM,CAAC,UAAS,SAAS,EAAE,SAAS;QACzD,IAAI,QAAQ,GAAG,aAAa,CAAC,SAAS,CAAC,CAAC;QAExC,IAAK,QAAQ,CAAC,MAAM,EAAG;YACtB,QAAQ,GAAG,QAAQ,CAAC,GAAG,CAAC,UAAU,GAAG;gBACpC,OAAO,gBAAgB,CAAC,IAAI,GAAG,GAAG,CAAC,CAAC;YACrC,CAAC,CAAC,CAAC;YAEH,QAAQ,CAAC,OAAO,CAAC,EAAE,CAAC,CAAC;YACrB,SAAS,IAAI,QAAQ,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC;SAClC;QAED,OAAO,SAAS,CAAC;IAClB,CAAC,EAAE,SAAS,CAAC,CAAC;IAEd,2BAA2B;IAC3B,IAAK,IAAI,CAAC,KAAK,CAAC,MAAM,EAAG;QACxB,SAAS,IAAI,KAAK,GAAG,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC;KAC5C;IAED,aAAa;IACb,IAAI,MAAM,GAAG,IAAI,CAAC,MAAM,CAAC;IACzB,SAAS,GAAG,MAAM,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC,MAAM,CAAC,UAAS,SAAS,EAAE,KAAK;QAC/D,OAAO,SAAS,IAAI,MAAM,CAAC,KAAK,CAAC,CAAC,CAAC,CAAC,KAAK,GAAG,EAAE,GAAG,KAAK,CAAC,CAAC,CAAC,EAAE,CAAC;IAC7D,CAAC,EAAE,SAAS,CAAC,CAAC;IAEd,OAAO,SAAS,CAAC;AAClB,CAAC;AAED;IACC,IAAI,UAAU,GAAG,QAAQ,CAAC,IAAI,CAAC,IAAI,CAAC,EAAE,CAAC;IACvC,OAAO,MAAM,CAAC,SAAS,CAAC,KAAK,CAAC,KAAK,CAAC,UAAU,EAAE,SAAS,CAAC,CAAC;AAC5D,CAAC;AAED;;;GAGG;AACH,aAAa,SAAS;IACrB,IAAI,OAAO,GAAG,IAAI,CAAC,IAAI,CAAC,EACvB,OAAO,CAAC;IAET,IAAK,SAAS,EAAG;QAChB,IAAK,OAAO,SAAS,KAAK,UAAU,EAAG;YACtC,OAAO,GAAG,CAAE,SAAS,CAAC,QAAQ,EAAE,CAAE,CAAC;SACnC;aAAM,IAAK,KAAK,CAAC,OAAO,CAAC,SAAS,CAAC,EAAG;YACrC,OAAO,GAAG,SAAS,CAAC;SACrB;aAAM,IAAK,OAAO,SAAS,KAAK,QAAQ,EAAG;YAC1C,OAAO,GAAG,CAAE,SAAS,CAAC,QAAQ,EAAE,CAAE,CAAC;SACpC;aAAM,IAAK,OAAO,SAAS,KAAK,QAAQ,EAAG;YAC3C,OAAO,GAAG,CAAE,SAAS,CAAE,CAAC;SACxB;aAAO;YACP,OAAO,GAAG,aAAa,CAAC,SAAS,EAAE,EAAE,CAAC,CAAC;SACvC;QACD,OAAO,CAAC,KAAK,GAAG,OAAO,CAAC,KAAK,CAAC,MAAM,CAAC,OAAO,CAAC,CAAC;KAC9C;IAED,OAAO,OAAO,CAAC,OAAO,CAAC,CAAC;AACzB,CAAC;AAED;;;;GAIG;AACH,eAAe,GAAG;IACjB,IAAI,OAAO,GAAG,IAAI,CAAC,IAAI,CAAC,EACvB,MAAM,GAAG,IAAI,CAAC,OAAO,CAAC,MAAM,CAAC,CAAC;IAE/B,MAAM,CAAC,MAAM,EAAE,GAAG,IAAI,EAAE,CAAC,CAAC;IAC1B,OAAO,CAAC,MAAM,GAAG,MAAM,CAAC;IAExB,OAAO,OAAO,CAAC,OAAO,CAAC,CAAC;AACzB,CAAC;AAED;;;;GAIG;AACH,iBAAiB,OAAO;IACvB,OAAO,GAAG,MAAM,CAAC;QAChB,IAAI,EAAE,EAAE;QACR,IAAI,EAAE,EAAE;QACR,KAAK,EAAE,EAAE;QACT,MAAM,EAAE,EAAE;KACV,EAAE,OAAO,IAAI,EAAE,CAAC,CAAC;IAElB,oEAAoE;IACpE,8CAA8C;IAC9C,IAAI,CAAC,GAAO,EAAE,CAAA;IACd,CAAC,CAAC,EAAE,GAAG,gBAAgB,CAAC,IAAI,CAAC,OAAO,CAAC,CAAA;IACrC,CAAC,CAAC,GAAG,GAAG,gBAAgB,CAAC,IAAI,CAAC,OAAO,CAAC,CAAA;IACtC,iBAAiB;IACjB,CAAC,CAAC,QAAQ,GAAG,QAAQ,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IACpC,CAAC,CAAC,KAAK,GAAG,KAAK,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IAC9B,CAAC,CAAC,GAAG,GAAG,GAAG,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IAC1B,CAAC,CAAC,KAAK,GAAG,KAAK,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC;IAE9B,OAAO,CAAC,CAAC;AACV,CAAC;AAED;;;;GAIG;AACH,eAAsB,IAAK;IAC1B,OAAO,OAAO,CAAC,EAAE,IAAI,EAAE,IAAI,EAAE,CAAC,CAAC;AAChC,CAAC;AAFD,sBAEC;AACD;;GAEG;AACH,MAAM,CAAC,KAAK,EAAE;IACb,KAAK,EAAC,UAAS,GAAG;QACjB,MAAM,CAAC,QAAQ,EAAE,GAAG,IAAI,EAAE,CAAC,CAAC;QAC5B,OAAO,KAAK,CAAC;IACd,CAAC;CACD,CAAC,CAAC"}