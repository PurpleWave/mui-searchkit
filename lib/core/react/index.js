function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./block"));
__export(require("./FastClick"));
__export(require("./LoadingComponent"));
__export(require("./RenderComponent"));
__export(require("./SearchkitComponent"));
__export(require("./SearchkitProvider"));
//# sourceMappingURL=index.js.map