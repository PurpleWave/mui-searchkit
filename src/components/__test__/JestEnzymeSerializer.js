let beautify = require('js-beautify')

module.exports = {
    test:function(wrapper) {
        return wrapper && wrapper.html
    },

    print(wrapper) {
        return beautify.html(wrapper.html(), { "indent_size": 2 })         
    }
}