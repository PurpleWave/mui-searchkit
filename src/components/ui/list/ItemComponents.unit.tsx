import * as React from "react"
import {mount} from "enzyme"

import {
  ItemComponent,
  CheckboxItemComponent,
} from "./ItemComponents"

import {
  block
} from "../../../"

import { fastClick} from "../../__test__/TestHelpers"

describe("ItemComponents", ()=> {

  beforeEach(()=> {
    this.bemBlocks = {
      container:block("sk-item-container").el,
      option:block("sk-item-option").el
    }
    this.onClick = jasmine.createSpy("toggleItem")


    this.props = {
      label:"Images", count:10, itemKey:"images",
      onClick: this.onClick,
      bemBlocks:this.bemBlocks, showCount:true
    }

  })

  it("should render and behave correctly", ()=> {

    this.wrapper = mount(
      <ItemComponent {...this.props}/>
    )
        
    expect(this.wrapper).toMatchSnapshot()
    this.wrapper.setProps({showCount:false, showCheckbox:true})

    expect(this.wrapper).toMatchSnapshot()
    
    this.wrapper.setProps({showCount:true, showCheckbox:false, count: undefined})

    expect(this.wrapper).toMatchSnapshot()


    this.wrapper.setProps({active:true})

    expect(this.wrapper.find(".sk-item-option").hasClass("is-active")).toBe(true)

    expect(this.onClick).not.toHaveBeenCalled()
    fastClick(this.wrapper.find(".sk-item-option"))
    expect(this.onClick).toHaveBeenCalled()

  })


  it("test default props for subclassed components", ()=> {
    expect(CheckboxItemComponent.defaultProps.showCheckbox).toBe(true)
    expect(ItemComponent.defaultProps.showCheckbox).toBe(false)
  })


})
