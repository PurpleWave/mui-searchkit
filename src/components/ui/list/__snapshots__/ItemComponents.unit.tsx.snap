// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponents should render and behave correctly 1`] = `
<div class="sk-item-option sk-item-container__item" data-qa="option" data-key="images">
  <div data-qa="label" class="sk-item-option__text">Images</div>
  <div data-qa="count" class="sk-item-option__count">10</div>
</div>
`;

exports[`ItemComponents should render and behave correctly 2`] = `
<div class="sk-item-option sk-item-container__item" data-qa="option" data-key="images"><input type="checkbox" data-qa="checkbox" readonly="" class="sk-item-option__checkbox">
  <div data-qa="label" class="sk-item-option__text">Images</div>
</div>
`;

exports[`ItemComponents should render and behave correctly 3`] = `
<div class="sk-item-option sk-item-container__item" data-qa="option" data-key="images">
  <div data-qa="label" class="sk-item-option__text">Images</div>
</div>
`;
